### DISCUSSION TITLE
currentMediaChanged 中调用 mediaInfo 有可能导致程序crash吗

### DISCUSSION THREAD

**Discussion by RAVER0306** [wang-bin/mdk-sdk#239] [Q&A]
查看当前堆栈似乎在 mediaInfo 之后有出现空指针，偶发

**Comment by wang-bin** [comment]
currentMediaChanged只是url变化，视频还没加载，mediaInfo是无效的，不过会回清除上一次的值，会不会crash看你访问了什么成员。要保证mediaInfo是当前视频的有效信息，必须在当前视频加载完之后调用，可以是prepare()的回调，也可以是onMediaStatus()里, 参考 https://github.com/wang-bin/mdk-examples/blob/master/Native/MediaInfoReader.cpp
